WebDriverAgentMac/WebDriverAgentLib/Commands/FBScreenshotCommands.m

```diff
@@ -39,4 +43,32 @@ + (NSArray *)routes
   return FBResponseWithObject(screenshot);
 }
 
++ (id<FBResponsePayload>)handleGetScreenshots:(FBRouteRequest *)request
+{
+  NSNumber *desiredId = request.arguments[@"displayId"];
```

Comment:
should catch an error to say the value must be number as the error, or accept the string value (converting the string to number if `desiredId` was not nil)? (the error itself was L53)
```
$ curl -X POST http://localhost:10100/wda/screenshots -d "{\"displayId\": \"2077749241\"}"
{
  "value" : {
    "error" : "unknown error",
    "message" : "-[__NSCFString isEqualToNumber:]: unrecognized selector sent to instance 0x6000002fc300",
    "traceback" : "(\n\t0   CoreFoundation                      0x00007fff34eae4d7 __exceptionPreprocess + 250\n\t1   libobjc.A.dylib                     0x00007fff6da275bf objc_exception_throw + 48\n\t2   CoreFoundation                      0x00007fff34f2d607 -[NSObject(NSObject) __retain_OA] + 0\n\t3   CoreFoundation                      0x00007fff34e12d2b ___forwarding___ + 1427\n\t4   CoreFoundation                      0x00007fff34e12708 _CF_forwarding_prep_0 + 120\n\t5   WebDriverAgentLib                   0x0000000105b6c015 +[FBScreenshotCommands handleGetScreenshots:] + 597\n\t6   WebDriverAgentLib                   0x0000000105bbf896 -[FBRoute_TargetAction mountRequest:intoResponse:] + 182\n\t7   WebDriverAgentLib                   0x0000000105bb8eb6 __37-[FBWebServer registerRouteHandlers:]_block_invoke + 854\n\t8   WebDriverAgentLib                   0x0000000105bb18bb -[RoutingHTTPServer handleRoute:withRequest:response:] + 219\n\t9   WebDriverAgentLib                   0x0000000105bb27b3 __72-[RoutingHTTPServer routeMethod:withPath:parameters:request:connection:]_block_invoke + 83\n\t10  libdispatch.dylib                   0x00007fff6eb76658 _dispatch_client_callout + 8\n\t11  libdispatch.dylib                   0x00007fff6eb827ab _dispatch_async_and_wait_invoke + 87\n\t12  libdispatch.dylib                   0x00007fff6eb76658 _dispatch_client_callout + 8\n\t13  libdispatch.dylib                   0x00007fff6eb81cab _dispatch_main_queue_callback_4CF + 936\n\t14  CoreFoundation                      0x00007fff34e717b1 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9\n\t15  CoreFoundation                      0x00007fff34e315b7 __CFRunLoopRun + 2028\n\t16  CoreFoundation                      0x00007fff34e3076e CFRunLoopRunSpecific + 462\n\t17  Foundation                          0x00007fff374cec48 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212\n\t18  WebDriverAgentLib                   0x0000000105bb7ae9 -[FBWebServer startServing] + 329\n\t19  WebDriverAgentRunner                0x0000000100bd8b72 -[UITestingUITests testRunner] + 82\n\t20  CoreFoundation                      0x00007fff34e1421c __invoking___ + 140\n\t21  CoreFoundation                      0x00007fff34e140c1 -[NSInvocation invoke] + 303\n\t22  XCTest                              0x0000000100282d3a __24-[XCTestCase invokeTest]_block_invoke_3 + 52\n\t23  XCTest                              0x0000000100282cee __24-[XCTestCase invokeTest]_block_invoke_2 + 297\n\t24  XCTest                              0x000000010031259a -[XCTMemoryChecker _assertInvalidObjectsDeallocatedAfterScope:] + 65\n\t25  XCTest                              0x000000010028d8ea -[XCTestCase assertInvalidObjectsDeallocatedAfterScope:] + 61\n\t26  XCTest                              0x0000000100282b82 __24-[XCTestCase invokeTest]_block_invoke.231 + 199\n\t27  XCTest                              0x00000001002f76d8 -[XCTestCase(XCTIssueHandling) _caughtUnhandledDeveloperExceptionPermittingControlFlowInterruptions:caughtInterruptionException:whileExecutingBlock:] + 179\n\t28  XCTest                              0x0000000100282645 -[XCTestCase invokeTest] + 1037\n\t29  XCTest                              0x0000000100284023 __26-[XCTestCase performTest:]_block_invoke_2 + 43\n\t30  XCTest                              0x00000001002f76d8 -[XCTestCase(XCTIssueHandling) _caughtUnhandledDeveloperExceptionPermittingControlFlowInterruptions:caughtInterruptionException:whileExecutingBlock:] + 179\n\t31  XCTest                              0x0000000100283f5a __26-[XCTestCase performTest:]_block_invoke.362 + 86\n\t32  XCTest                              0x0000000100308b9f +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 220\n\t33  XCTest                              0x00000001002837c7 -[XCTestCase performTest:] + 695\n\t34  XCTest                              0x00000001002d56da -[XCTest runTest] + 57\n\t35  XCTest                              0x000000010027d035 __27-[XCTestSuite performTest:]_block_invoke + 329\n\t36  XCTest                              0x000000010027c856 __59-[XCTestSuite _performProtectedSectionForTest:testSection:]_block_invoke + 24\n\t37  XCTest                              0x0000000100308b9f +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 220\n\t38  XCTest                              0x0000000100308ab0 +[XCTContext runInContextForTestCase:block:] + 52\n\t39  XCTest                              0x000000010027c80d -[XCTestSuite _performProtectedSectionForTest:testSection:] + 148\n\t40  XCTest                              0x000000010027cb11 -[XCTestSuite performTest:] + 290\n\t41  XCTest                              0x00000001002d56da -[XCTest runTest] + 57\n\t42  XCTest                              0x000000010027d035 __27-[XCTestSuite performTest:]_block_invoke + 329\n\t43  XCTest                              0x000000010027c856 __59-[XCTestSuite _performProtectedSectionForTest:testSection:]_block_invoke + 24\n\t44  XCTest                              0x0000000100308b9f +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 220\n\t45  XCTest                              0x0000000100308ab0 +[XCTContext runInContextForTestCase:block:] + 52\n\t46  XCTest                              0x000000010027c80d -[XCTestSuite _performProtectedSectionForTest:testSection:] + 148\n\t47  XCTest                              0x000000010027cb11 -[XCTestSuite performTest:] + 290\n\t48  XCTest                              0x00000001002d56da -[XCTest runTest] + 57\n\t49  XCTest                              0x000000010027d035 __27-[XCTestSuite performTest:]_block_invoke + 329\n\t50  XCTest                              0x000000010027c856 __59-[XCTestSuite _performProtectedSectionForTest:testSection:]_block_invoke + 24\n\t51  XCTest                              0x0000000100308b9f +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 220\n\t52  XCTest                              0x0000000100308ab0 +[XCTContext runInContextForTestCase:block:] + 52\n\t53  XCTest                              0x000000010027c80d -[XCTestSuite _performProtectedSectionForTest:testSection:] + 148\n\t54  XCTest                              0x000000010027cb11 -[XCTestSuite performTest:] + 290\n\t55  XCTest                              0x00000001002d56da -[XCTest runTest] + 57\n\t56  XCTest                              0x00000001003258b5 __44-[XCTTestRunSession runTestsAndReturnError:]_block_invoke_2 + 148\n\t57  XCTest                              0x0000000100308b9f +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 220\n\t58  XCTest                              0x0000000100308ab0 +[XCTContext runInContextForTestCase:block:] + 52\n\t59  XCTest                              0x000000010032581a __44-[XCTTestRunSession runTestsAndReturnError:]_block_invoke + 111\n\t60  XCTest                              0x000000010032599b __44-[XCTTestRunSession runTestsAndReturnError:]_block_invoke.95 + 96\n\t61  XCTest                              0x00000001002a3cb8 -[XCTestObservationCenter _observeTestExecutionForBlock:] + 325\n\t62  XCTest                              0x00000001003255e0 -[XCTTestRunSession runTestsAndReturnError:] + 615\n\t63  XCTest                              0x0000000100260a7e -[XCTestDriver _runTests] + 466\n\t64  XCTest                              0x0000000100304b82 _XCTestMain + 108\n\t65  WebDriverAgentRunner-Runner         0x0000000100003d41 _XCTRunnerRunTests + 25\n\t66  WebDriverAgentRunner-Runner         0x0000000100003d28 _XCTRunnerRunTests + 0\n\t67  libdyld.dylib                       0x00007fff6ebcfcc9 start + 1\n\t68  ???                                 0x0000000000000005 0x0 + 5\n)"
  },
  "sessionId" : null
}%
```

Reply:
The documentation explicitly says the parameter type must be a string, so I'd say this is expected